middleware.ts

```diff
@@ -1,22 +1,73 @@
 import { NextRequest, NextResponse } from 'next/server'
 
+import { createQueryString } from './lib/routing';
+
+const envIdRegex = /(?<envId>[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12})(?<remainingUrl>.*)/
+
 export const middleware = (request: NextRequest) => {
-  // /articles will be shown in as URL but request will be against fully defined URL path
+  let currentEnvId = request.cookies.get('currentEnvId')?.value;
+
+  if (!currentEnvId) {
+    currentEnvId = process.env.NEXT_PUBLIC_KONTENT_ENVIRONMENT_ID;
+  }
+
+  if (!currentEnvId) {
+    throw new Error("Missing 'NEXT_PUBLIC_KONTENT_ENVIRONMENT_ID' environment variable.");
+  }
+
+
+  const regexResult = request.nextUrl.pathname.match(envIdRegex);
+  const routeEnvId = regexResult?.groups?.envId
+  const remainingUrl = regexResult?.groups?.remainingUrl;
+
+  if (routeEnvId) {

```

Comment:
What would you think about a function call like `handleExplicitProjectRoute(routeEnvId, remainingUrl)` instead of the body of the `if`?